test/e2e/pd.go

```diff
@@ -0,0 +1,217 @@
+limitations under the License.
+*/
+
+package e2e
+
+import (
+	"fmt"
+	"os/exec"
+	"strings"
+	"time"
+
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/api"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/client"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/util"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("PD", func() {
+	var (
+		c         *client.Client
+		host0Name string
+		host1Name string
+	)
+
+	BeforeEach(func() {
+		var err error
+		c, err = loadClient()
+		Expect(err).NotTo(HaveOccurred())
+
+		nodes, err := c.Nodes().List()
+		expectNoError(err, "Failed to list nodes for e2e cluster.")
+		Expect(len(nodes.Items) >= 2)
+
+		host0Name = nodes.Items[0].ObjectMeta.Name
+		host1Name = nodes.Items[1].ObjectMeta.Name
+	})
+
+	It("should schedule a pod w/ a RW PD, remove it, then schedule it on another host", func() {
+		podClient := c.Pods(api.NamespaceDefault)
+		diskName := fmt.Sprintf("e2e-%d", time.Now().Unix())
+
+		host0Pod := testPDPod(diskName, host0Name, false)
+		host1Pod := testPDPod(diskName, host1Name, false)
+
+		By("creating PD")
+		err := createPD(diskName, testContext.gceConfig.Zone)
+		expectNoError(err, fmt.Sprintf("Error creating PD: %v", err))
+
+		defer func() {
+			By("cleaning up PD-RW test environment")
+			defer GinkgoRecover()
+			// Teardown pods, PD. Ignore errors. Teardown should do nothing
+			// unless test failed.
+			podClient.Delete(host0Pod.Name)
+			podClient.Delete(host1Pod.Name)
+			detachPD(host0Name, diskName, testContext.gceConfig.Zone)
+			detachPD(host1Name, diskName, testContext.gceConfig.Zone)
+			deletePD(diskName, testContext.gceConfig.Zone)
+		}()
+
+		By("submitting host0Pod to kubernetes")
+		_, err = podClient.Create(host0Pod)
+		expectNoError(err, fmt.Sprintf("Failed to create host0Pod: %v", host0Pod.Name, err))
+
+		By("waiting up to 60 seconds for host0Pod to start running")
+		err = waitForPodRunning(c, host0Pod.Name, 60*time.Second)
+		expectNoError(err, fmt.Sprintf("host0Pod not running after 60 seconds: %v", err))
+
+		By("deleting host0Pod")
+		err = podClient.Delete(host0Pod.Name)
+		expectNoError(err, fmt.Sprintf("Failed to delete host0Pod: %v", host0Pod.Name, err))
+
+		By("submitting host1Pod to kubernetes")
+		_, err = podClient.Create(host1Pod)
+		expectNoError(err, fmt.Sprintf("Failed to create host1Pod: %v", host1Pod.Name, err))
+
+		By("waiting up to 60 seconds for host1Pod to start running")
+		err = waitForPodRunning(c, host1Pod.Name, 60*time.Second)
+		expectNoError(err, fmt.Sprintf("host1Pod not running after 60 seconds: %v", err))
+
+		By("deleting host1Pod")
+		err = podClient.Delete(host1Pod.Name)
+		expectNoError(err, fmt.Sprintf("Failed to delete host1Pod: %v", host1Pod.Name, err))
+
+		return
+	})
+
+	It("should schedule a pod w/ a readonly PD on two hosts, then remove both.", func() {
+		podClient := c.Pods(api.NamespaceDefault)
+		diskName := fmt.Sprintf("e2e-%d", time.Now().Unix())
```

Comment:
Just for consistency with other tests, I'd use `randomSuffix` here as a uniquifier.